### DISCUSSION TITLE
Let's get rid of useless tooltips that only display what's already shown

### DISCUSSION THREAD

**Discussion by sszigeti** [godotengine/godot-proposals#6235] [Editor]
If you hover the mouse over the FileSystem entries or the list of methods, the tooltips are simply repeating the content of whatever item I'm hovering, and thus giving no useful information but obscuring the UI. This can also be observed in the Inspector, if you hover over collapsible section headers like the "Anchor", "Margin", "Grow Direction".

Would it be possible for the editor tooltip to only appear if it's actually contributing any useful information (that's not already being shown by its owner/trigger node)?

**Comment by Lielay9** [comment]
I could've sworn there were settings to adjust tooltips. Instead got hit with the irony:

![image](https://user-images.githubusercontent.com/40064911/216780902-b6f5d805-d957-48aa-995c-6dc66440f237.png)

Additional places where the tooltips are redundant:
- FileSystem (tree)
- FileDialog (recent, favorites and directories & files)
- Quick Open ... (list)
- Groups (list + no tooltips for buttons)
- History (list)
- Debugger - Monitors (Monitor)
- ProjectSettings - General (tree)
- ProjectSettings - Input Map (action names and values)
- ProjectSettings - Autoload (name and path + no tooltips for the buttons)
- Command Palette (tree)
- EditorSettings - General (tree)
- EditorSettings - Shortcuts (tree + no tooltips for the buttons)
- About Godot - Third-party Licenses (tree)
- Orphan Resource Explorer (tree -> resource and owns)
- Edit Build Configuration Profile (tree)
- Manage Editor Feature Profiles (tree)

There's more but I got bored and seems that the main culprit is the tree-types. Maybe they should only show tooltip if they're not the default value. Additionally some of the tooltips could be made more useful. The FileSystem tooltip for example could show the whole path instead of the name of the file (better, but not sure if desirable).

**Comment by dalexeev** [comment]
This hint is needed if the text of the list/tree item does not fit.

**Reply by WhalesState** [reply]
`Tree` is using a `ScrollContainer` by default, what about `ItemList`? what about horizontal scrolling when it doesn't fit ?

**Comment by CodingOctocat** [comment]
Agree, only provide ToolTip if content is too long(not fully displayed) or if meaningful help is required.

**Reply by WhalesState** [reply]
This will break compatibility, since users will not be able to set a custom tooltip for `Tree` items or `ItemList` items.

**Reply by aXu-AP** [reply]
I think it won't be compatibility breaking if user provided tooltips will be always shown.

**Reply by WhalesState** [reply]
AThousandShips is already working on that. https://github.com/godotengine/godot/pull/97158
